File: python/demos/demo_stanley.py
#!/usr/bin/env python3
"""Stanley demo."""

from __future__ import annotations

from demo_common import run_local

# Imported from the submodule directly, mirroring demo_pure_pursuit.py's
# precedent (the package re-export lives in
# navigation/local_planning/tracking/__init__.py).
from navigation.local_planning.tracking.stanley import Stanley

if __name__ == "__main__":
    run_local("stanley", Stanley)

File: cpp/demos/demo_stanley.cpp
#include "demo_common.hpp"
#include "navigation/local_planning/tracking/stanley.hpp"

int main(int argc, char** argv) {
  try {
    return demo::run_local(argc, argv, "stanley", [](const navigation::core::ParamSet& p) {
      return navigation::local_planning::StanleyPlanner(p);
    });
  } catch (const std::exception& e) {
    std::cerr << "error: " << e.what() << "\n";
    return 1;
  }
}
